core/src/main/java/org/apache/iceberg/rest/HTTPClient.java

```diff
@@ -448,6 +443,43 @@ static HttpRequestInterceptor loadInterceptorDynamically(
     return instance;
   }
 
+  static HttpClientConnectionManager getConnectionManager(
+      Map<String, String> properties, ConnectionConfig connectionConfig) {
+    PoolingHttpClientConnectionManagerBuilder connectionManagerBuilder =
+        PoolingHttpClientConnectionManagerBuilder.create();
+    if (connectionConfig != null) {
+      connectionManagerBuilder.setDefaultConnectionConfig(connectionConfig);
+    }
+    return connectionManagerBuilder
+        .useSystemProperties()
+        .setMaxConnTotal(Integer.getInteger(REST_MAX_CONNECTIONS, REST_MAX_CONNECTIONS_DEFAULT))
+        .setMaxConnPerRoute(
+            PropertyUtil.propertyAsInt(
+                properties, REST_MAX_CONNECTIONS_PER_ROUTE, REST_MAX_CONNECTIONS_PER_ROUTE_DEFAULT))
+        .build();
+  }
+
+  @VisibleForTesting
+  static ConnectionConfig getConnectionConfig(Map<String, String> properties) {
+    Long connectionTimeout =
+        PropertyUtil.propertyAsNullableLong(properties, REST_CONNECTION_TIMEOUT);
+    Integer socketTimeout = PropertyUtil.propertyAsNullableInt(properties, REST_SOCKET_TIMEOUT);
```

Comment:
why is one Long and the other Integer?

Reply:
SocketTimeout is an int type in the apache closeable httpclient whereas connection timeout is a long